tests/data-type.multi.value.spec.js

```diff
@@ -0,0 +1,142 @@
+const expect = require('chai').expect;
+const MultiValue = require('../src/data_type/multi.value');
+const Image = require('../src/data_type/image');
+const ImageJpeg = require('../src/data_type/image.jpeg');
+const ImagePng = require('../src/data_type/image.png');
+
+const getTestValues = () => [
+  new ImageJpeg('image_1'),
+  new ImagePng('image_2'),
+  'test string',
+  { name: 'test object' },
+  ['test', 'array'],
+  new MultiValue([
+    new ImageJpeg('image_3'),
+    new ImagePng('image_4'),
+    'test string 2',
+    { name: 'test object 2' },
+    ['test', 'array 2'],
+    new MultiValue([
+      new ImageJpeg('image_5'),
+      new ImagePng('image_6'),
+      'test string 3',
+      { name: 'test object 3' },
+      ['test', 'array 3'],
+    ]),
+  ]),
+];
+
+describe('MultiValue', () => {
+  context('#getItems', () => {
+    it('should return provided array of items', () => {
+      const multiValue = new MultiValue(getTestValues());
+      expect(multiValue.getItems()).to.eql(getTestValues());
+    });
+  });
+  context('#applyFilters', () => {
+    it('should apply filters to all multi values', () => {
+      const multiValue = new MultiValue(getTestValues());
+      expect(multiValue.getItems().length).to.equal(6);
+
+      // Apply filters.
+      multiValue
+        .filterInstance(ImageJpeg)
```

Comment:
This really would be easier to follow if the methods had names like _addAllowedType_, or _allowType_ or something.

What the filter really does is actually the opposite of what you'd expect from _filterType(Image)_.  I think you'd expect that to remove _Image_, rather than allow it.

Reply:
Renamed to `allow...()` in d6edd2d